receiver/mongodbatlasreceiver/internal/metric_conversion.go

```diff
@@ -19,15 +19,81 @@ func processMeasurements(
 	var errs error
 
 	for _, meas := range measurements {
-		err := metadata.MeasurementsToMetric(mb, meas, false)
+		err := metadata.MeasurementsToMetric(mb, meas)
 		if err != nil {
 			errs = multierr.Append(errs, err)
 		}
 	}
 
+	err := calculateTotalMetrics(mb, measurements)
+	if err != nil {
+		errs = multierr.Append(errs, err)
+	}
 	if errs != nil {
 		return fmt.Errorf("errors occurred while processing measurements: %w", errs)
 	}
 
 	return nil
 }
+
+func calculateTotalMetrics(
+	mb *metadata.MetricsBuilder,
+	measurements []*mongodbatlas.Measurements,
+) error {
+	var err error
+	dptTotalMeasCombined := false
+	var dptTotalMeas *mongodbatlas.Measurements
+
+	for _, meas := range measurements {
+		switch meas.Name {
+		case "DISK_PARTITION_THROUGHPUT_READ":
+			fallthrough
+		case "DISK_PARTITION_THROUGHPUT_WRITE":
+			if dptTotalMeas == nil {
+				dptTotalMeas = cloneMeasurement(meas)
+				dptTotalMeas.Name = "DISK_PARTITION_THROUGHPUT_TOTAL"
+				continue
+			}
+
+			// Combine data point values with matching timestamps
+			for j, totalMeas := range dptTotalMeas.DataPoints {
+				if totalMeas.Timestamp != meas.DataPoints[j].Timestamp ||
```

Comment:
Will probably need to test to make sure matching timestamps is sufficent enough to combine datapoints. I may have to poke around the API to make sure that is a valid assumption at this step.

Reply:
@schmikei I have done some manual spot checking and it has looked good so far (multiple datapoints coming through in a single collection, and the totals being created properly).

Outside of that I'll leave it up to you. I can rip this out if need be, as ultimately it was a "nice to have" although hopefully it doesn't come to it.